File: angles_testing.py
import math

from matplotlib import pyplot as plt


def earthBuffers():
    latitudeBands = 120
    longitudeBands = 120
    radius = 5
    vertexPositionData = []
    normalData = []
    textureCoordData = []

    for latNumber in range(latitudeBands):
        theta = latNumber*math.pi / latitudeBands
        sinTheta = math.sin(theta)
        cosTheta = math.cos(theta)
        for longNumber in range(longitudeBands):
            phi = longNumber*2*math.pi / longitudeBands
            sinPhi = math.sin(phi)
            cosPhi = math.cos(phi)
            x = cosPhi * sinTheta
            y = cosTheta
            z = sinPhi * sinTheta
            u = 1 - (longNumber / longitudeBands)
            v = 1 - (latNumber / latitudeBands)
            normalData.append(x)
            normalData.append(y)
            normalData.append(z)
            textureCoordData.append(u)
            textureCoordData.append(v)
            vertexPositionData.append(radius * x)
            vertexPositionData.append(radius * y)
            vertexPositionData.append(radius * z)

    print (vertexPositionData)
    print (normalData)
    print (textureCoordData)

    plt.plot(range(len(normalData)//3), [x for i,x in enumerate(normalData) if i%3==0])
    plt.plot(range(len(normalData)//3), [x for i,x in enumerate(normalData) if i%3==1])
    plt.plot(range(len(normalData)//3), [x for i,x in enumerate(normalData) if i%3==2])
    plt.show()

    plt.plot(range(len(textureCoordData)//2), [x for i,x in enumerate(textureCoordData) if i%2==0])
    plt.plot(range(len(textureCoordData)//2), [x for i,x in enumerate(textureCoordData) if i%2==1])
    plt.show()

    plt.plot(range(len(vertexPositionData)//3), [x for i,x in enumerate(vertexPositionData) if i%3==0])
    plt.plot(range(len(vertexPositionData)//3), [x for i,x in enumerate(vertexPositionData) if i%3==1])
    plt.plot(range(len(vertexPositionData)//3), [x for i,x in enumerate(vertexPositionData) if i%3==2])
    plt.show()

earthBuffers()

File: src/static/js/angles_testing.js
        function setupEarthBuffers() {
            // splits Earth into 60 lat and long bands, can increase for greater detail
            var latitudeBands = 60; 
            var longitudeBands = 60;
            var radius = 5;
            var vertexPositionData = [];
            var normalData = [];
            var textureCoordData = [];

            for (var latNumber = 0; latNumber <= latitudeBands; latNumber++) {
                var theta = latNumber*Math.PI / latitudeBands;
                var sinTheta = Math.sin(theta);
                var cosTheta = Math.cos(theta);
                for (var longNumber = 0; longNumber <= longitudeBands; longNumber++) {
                    var phi = longNumber*2*Math.PI / longitudeBands;
                    var sinPhi = Math.sin(phi);
                    var cosPhi = Math.cos(phi);
                    var x = cosPhi * sinTheta;
                    var y = cosTheta;
                    var z = sinPhi * sinTheta;
                    var u = 1 - (longNumber / longitudeBands);
                    var v = 1 - (latNumber / latitudeBands);
                    normalData.push(x);
                    normalData.push(y);
                    normalData.push(z);
                    textureCoordData.push(u);
                    textureCoordData.push(v);
                    vertexPositionData.push(radius * x);
                    vertexPositionData.push(radius * y);
                    vertexPositionData.push(radius * z);
                }
            }

        }